opal/mca/threads/qthreads/threads_qthreads_mutex.h

```diff
@@ -108,9 +110,24 @@ static inline void opal_mutex_create(struct opal_mutex_t *m)
 #endif
 }
 
+static inline void opal_mutex_recursive_create(struct opal_mutex_t *m)
+{
+    opal_threads_ensure_init_qthreads();
+
+    opal_atomic_lock_init(&m->m_lock, 0);
+    opal_atomic_lock_init(&m->m_lock_atomic, 0);
+    m->m_recursive = 1;
```

Comment:
Shouldn't there be some check in `opal_mutex_trylock` whether `m_recursive == 1` and which thread owns the mutex?